### PR TITLE
Fix PostContent block rendering inside templates

### PR DESCRIPTION
**Pull request by youknowriad**
we have a conceptual issue in the rendering of FSE templates in relation to block context.

 - For "postId" and "postType" context values, we were using the global $post object (which make sense for blocks like core/post-title used inside the post content of posts for instance.

 - But when rendering templates, thee $post global is equivalent to the template and templates in general include a `core/post-content` block. This means we get an infinite loop. Post content inside the template renders the the content of the template itself.

This  is an attempt to fix by excluding the wp_template from that  context value (there might be a better way in the future).

### CODE REVIEW

File: lib/compat.php [hunk 1]
```diff
@@ -464,8 +463,8 @@ function gutenberg_inject_default_block_context( $args ) {
 			}
 
 			// Inject the query context if not done by Core.
-			$needs_query = empty( $block_type->uses_context ) && in_array( 'query', $block_type->uses_context, true );
-			if ( ! isset( $block->context['query'] ) && $needs_query ) {
+			$needs_query = ! empty( $block_type->uses_context ) && in_array( 'query', $block_type->uses_context, true );
+			if ( ! isset( $block->context['query'] ) && $needs_query && ! isset( $block->context['query'] ) ) {
```
Comment (ellatrix) [inline lib/compat.php:467]: Not sure if this needs the extra checks you added above. I guess the tests will tell.

Comment (youknowriad) [reply lib/compat.php]: it doesn't IMO

Comment (youknowriad) [reply lib/compat.php]: As you can see, the old filter I removed was unsetting just postId and postType. (it was the same code but for the old context implementation) but now that I think about it, I think we need to revert the removal of that filter to solve the issues here..

Comment (youknowriad) [reply lib/compat.php]: basically we need to revert this removal https://github.com/WordPress/gutenberg/pull/27011/files#diff-9f399722fe93f7131d23a3944e030b88420d66461c7b776e7b153bafd3ae7100L249-L273

Comment (ellatrix) [reply lib/compat.php:467]: Ok, I removed my commits. There's still one instance of `in_array` that needs to be guarded for empty values.

Comment (noahtallen) [reply lib/compat.php:467]: I added back the filter that was removed and fixed the other instance of `in_array`

Comment (noahtallen) [reply lib/compat.php:467]: This check for `isset( $block->context['query'] )` is also redundant (it's already earlier in the same line), so I removed that too

File: lib/templates-sync.php [hunk 2]
```diff
@@ -166,7 +166,7 @@ function gutenberg_clear_synchronize_last_checks() {
  * @param int     $post_id ID of the deleted post.
  * @param WP_Post $post WP_Post instance of the deleted post.
  */
-function gutenberg_clear_synchronize_last_checks_after_delete( $postid, $post ) {
+function gutenberg_clear_synchronize_last_checks_after_delete( $post_id, $post ) {
```
Comment (noahtallen) [inline lib/templates-sync.php:169]: this was causing a phpcs issue, just making sure we have it here to test that phpunit is good

**Review by noahtallen — APPROVED** [review]
Approving this because:
1. it fixes the logic issue
2. it fixes the phpcs issue
3. it solves the issue without breaking the site editor
4. it solves the issue without showing an error.

So I'll go ahead and merge this! 🎉 I think further improvements to this could be done later. My main goal is to just get it working (particularly the tests)

**Pull request merged**